### PR TITLE
POK-134: refactored backend build and test process.

### PR DESCRIPTION
**Pull request by frei-sven**
- Setup backend test coverage reporting to sonar cloud.
- Removed implicit usings from the project files to avoid random build errors.

### CODE REVIEW

File: PokeOneWeb.Shared/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -6,5 +6,10 @@ public static bool EqualsExact(this string str, string other)
         {
             return str.Equals(other, StringComparison.Ordinal);
         }
+
+        public static bool EqualsExact2(this string str, string other)
```
Comment (Finrod-Amandil) [inline PokeOneWeb.Shared/Extensions/StringExtensions.cs]: Why do you need this duplicate?

Comment (frei-sven) [reply PokeOneWeb.Shared/Extensions/StringExtensions.cs]: Removed, was for testing purpose for the code coverage on new lines only.

File: .github/workflows/pokeone-api-pr.yaml [hunk 2]
```diff
@@ -23,17 +23,17 @@ jobs:
...
-    - name: Build
-      run: dotnet build PokeOneWeb --configuration Release --no-restore
+    - name: Build & Install Dependencies
+      run: dotnet build PokeOneWeb --configuration Release
     - name: Test
-      run: dotnet test --no-restore --verbosity normal
-
-    #- name: Unit-Test Api
-    #  run: dotnet test PokeOneWeb --no-build --no-restore --verbosity normal -p:CollectCoverage=true -p:CoverletOutputFormat=opencover
+      run: dotnet test --collect "XPlat Code Coverage" --results-directory TestResults/ --logger "trx;LogFileName=unittests.trx" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover
```
Comment (Finrod-Amandil) [inline .github/workflows/pokeone-api-pr.yaml:36]: For test coverage results I would suggest to **exclude** the following directories:
- all test projects (only measure coverage on production code, test code is anway always 100% covered)
- all generated code. This includes the DB migrations and snapshots, located in PokeOneWeb.Data/Migrations.

**Review by Finrod-Amandil — APPROVED** [review]

File: .github/workflows/pokeone-api-pr.yaml (hunk 2 above)
Comment (frei-sven) [reply .github/workflows/pokeone-api-pr.yaml:36]: Good point, as far as I saw it the Test projects are excluded by default. And the Migration files are excluded by the project settings on SonarCloud. Files that are marked auto-generated are overall excluded, but I think some Migration files are not marked liked this automatically (as far as I saw).
I rather configure such things on SonarCloud directly so we do not bloat our pipelines. But this is just my preference. We can also change this in the future.

**Pull request merged**